.github/workflows/markdownlint.yml

```diff
@@ -0,0 +1,51 @@
+name: Run markdownlint
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**/*.md'
+      - '!translated/**'
+      - '!versions/**'
+  workflow_dispatch: # Manual run
+
+concurrency:
+  group: markdownlint-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  markdownlint:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - id: checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - id: changes
+        uses: tj-actions/changed-files@v44
+        with:
+          files: |
+            **/*.md
+            !translated/**
+            !versions/**
+
+      - id: node
+        if: steps.changes.outputs.any_changed == 'true'
```

Comment:
I would just always lint the entire thing, we arent too bothered about speed here. Keep it simple.

Reply:
Some translations will be unlintable based on the way the translations are patched in by Crowdin, not nessecarily about speed. I agree that we should lint versioned pages if they're edited (just not translation pages from Crowdin)

Reply:
Does linting (aka modifying) translated files cause problems in Crowdin? That's my only concern

Reply:
> Does linting (aka modifying) translated files cause problems in Crowdin? That's my only concern
No, its more the other way round, the linter would error out pretty much instantly with most of the non-latin languages (chinese, etc.), I had a try and it just doesn't support them very well.

Reply:
> Even unchanged files? See https://github.com/FabricMC/fabric-docs/pull/101#discussion_r1597707214

A good point was brought up though, if we change lint rules, it'd enforce us to change the old pages :+1:

Reply:
tldr:

- Lint all files.
- Ignore anything in versions/<version>/translated/
- Ignore anything in translated/